### PR TITLE
Update na_ontap_lun.py

### LINKED ISSUE #55: na_ontap_lun.py: Wrong size starting from 450g
##### SUMMARY
When creating a LUN >= 450g ONTAP doesn't calculate the size 1024 based:

```
supernetapp::> lun create -vserver Test_SC1 -volume ultraLun -lun sql01.lun -ostype linux -size 450g

Created a LUN of size 450.1g (483247783936)
```

So when running the ansible-playbook a second time you'll get an error, that the script can't shrink the LUN.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
```
na_ontap_lun
```

##### ANSIBLE VERSION
```
2.9.12
```

##### CONFIGURATION
```
-
```

##### OS / ENVIRONMENT
Linux version 4.19.0-8-amd64 (debian-kernel@lists.debian.org) (gcc version 8.3.0 (Debian 8.3.0-6)) #1 SMP Debian 4.19.98-1+deb10u1 (2020-04-27)

VM

##### STEPS TO REPRODUCE
Just create a lun with **netapp.ontap.na_ontap_lun** size >= 450GB.

##### EXPECTED RESULTS
When run a second time, lun result is **ok**.

##### ACTUAL RESULTS
```
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~root && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp `"&& mkdir "` echo /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177 `" && echo ansible-tmp-1599651978.7138655-18688-206661629417177="` echo /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177 `" ) && sleep 0'
Using module file /root/.ansible/collections/ansible_collections/netapp/ontap/plugins/modules/na_ontap_lun.py
<127.0.0.1> PUT /root/.ansible/tmp/ansible-local-186831lqdpgcm/tmpis00dlfa TO /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177/AnsiballZ_na_ontap_lun.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177/ /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177/AnsiballZ_na_ontap_lun.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177/AnsiballZ_na_ontap_lun.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1599651978.7138655-18688-206661629417177/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_netapp.ontap.na_ontap_lun_payload_wnpnfpga/ansible_netapp.ontap.na_ontap_lun_payload.zip/ansible_collections/netapp/ontap/plugins/modules/na_ontap_lun.py", line 344, in resize_lun
  File "/usr/local/lib/python3.7/dist-packages/netapp_lib/api/zapi/zapi.py", line 301, in invoke_successfully
    raise NaApiError(code, msg)
netapp_lib.api.zapi.zapi.NaApiError: NetApp API failed. Reason - 9033:Reducing LUN size without coordination with the host system may cause permanent data loss or corruption. Use the force flag to allow LUN size reduction.
failed: [localhost] (item={'lun': 'sql01.lun', 'volume': 'ultraLun', 'vserver': 'Test_SC1', 'size': 800, 'size_unit': 'gb', 'ostype': 'linux', 'path': '/vol/ultraLun/sql01.lun', 'igroup': 'igroup01'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "cert_filepath": null,
            "feature_flags": {},
            "flexvol_name": "ultraLun",
            "force_remove": false,
            "force_remove_fenced": false,
            "force_resize": false,
            "hostname": "XXX",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "name": "sql01.lun",
            "ontapi": null,
            "ostype": "linux",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "size": 800,
            "size_unit": "gb",
            "space_allocation": false,
            "space_reserve": true,
            "state": "present",
            "use_rest": "auto",
            "username": "admin",
            "validate_certs": false,
            "vserver": "Test_SC1"
        }
    },
    "item": {
        "igroup": "igroup01",
        "lun": "sql01.lun",
        "ostype": "linux",
        "path": "/vol/ultraLun/sql01.lun",
        "size": 800,
        "size_unit": "gb",
        "volume": "ultraLun",
        "vserver": "Test_SC1"
    },
    "msg": "Error resizing lun /vol/ultraLun/sql01.lun: NetApp API failed. Reason - 9033:Reducing LUN size without coordination with the host system may cause permanent data loss or corruption. Use the force flag to allow LUN size reduction."
}
```

### PR DESCRIPTION
**Pull request by AwesomeNameMan**
Added the argument 'use_exact_size', see issue 55.

##### SUMMARY
Include "Fixes #55" 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
na_ontap_lun

##### ADDITIONAL INFORMATION
```
See Issue #55 
```

### CODE REVIEW

File: ansible_collections/netapp/ontap/plugins/modules/na_ontap_lun.py [hunk 1]
```diff
@@ -184,6 +190,7 @@ def __init__(self):
             ostype=dict(required=False, type='str', default='image'),
             space_reserve=dict(required=False, type='bool', default=True),
             space_allocation=dict(required=False, type='bool', default=False),
+            use_exact_size=dict(required=False, type='bool', default=False),
```
Comment (lonico) [inline ansible_collections/netapp/ontap/plugins/modules/na_ontap_lun.py:193]: default=True

**Review by lonico — CHANGES_REQUESTED** [review]
Good for me.

I was wondering is we should ignore any size change within 1% for instance.  And make this a parameter too.

But what you're doing here is good.

**Comment by carchi8py** [comment]
Look like ansible-test is finding a whitespace issue that needs to be fixed.

ERROR: Found 1 pep8 issue(s) which need to be resolved:
See documentation for help: https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/pep8.html
ERROR: plugins/modules/na_ontap_lun.py:109:1: W293: blank line contains whitespace

**Comment by AwesomeNameMan** [comment]
> 
> 
> Look like ansible-test is finding a whitespace issue that needs to be fixed.
> 
> ERROR: Found 1 pep8 issue(s) which need to be resolved:
> See documentation for help: https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/pep8.html
> ERROR: plugins/modules/na_ontap_lun.py:109:1: W293: blank line contains whitespace

Maybe not the best idea to use the github editor on mobile, will do the check tomorrow. Python Indentation can be fun :-).

**Comment by AwesomeNameMan** [comment]
Sorry for the delay, the sanity tests have now been successfully completed and everything should fit.

**Comment by AwesomeNameMan** [comment]
Still waiting for the review to happen. Is there something else I can do here?

File: ansible_collections/netapp/ontap/changelogs/fragments/20.10.0.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - na_ontap_lun - ``use_exact_size`` to create a lun with the exact given size so that the lun is not rounded up
```
Comment (lonico) [inline ansible_collections/netapp/ontap/changelogs/fragments/20.10.0.yaml:2]: can you revert this change?   You should not modify this file, and you already created a fragments/github-56.yaml

**Review by lonico — CHANGES_REQUESTED** [review]
Can you revert changelog.yaml and 20.10.0.yml

File: ansible_collections/netapp/ontap/changelogs/changelog.yaml [hunk 3]
```diff
@@ -849,3 +849,10 @@ releases:
       name: na_ontap_metrocluster
       namespace: ''
     release_date: '2020-09-02'
+  20.10.0:
```
[the remark this answers is not in the document]
Comment (AwesomeNameMan) [reply ansible_collections/netapp/ontap/changelogs/changelog.yaml]: Reverted.

File: ansible_collections/netapp/ontap/changelogs/fragments/20.10.0.yaml (hunk 2 above)
Comment (AwesomeNameMan) [reply ansible_collections/netapp/ontap/changelogs/fragments/20.10.0.yaml]: Deleted the file.

**Review by lonico — APPROVED** [review]
